devops/ansible/roles/girder/library/girder.py

```diff
@@ -942,10 +950,15 @@ def update(self, _id, body, **kwargs):
         if _id in self.resources:
             current = self.resources[_id]
             # if body is a subset of current we don't actually need to update
-            if set(body.items()) <= set(current.items()):
-                return current
-            else:
-                return self.__apply(_id, self.client.put, body, **kwargs)
+            try:
+                if set(body.items()) <= set(
```

Comment:
@manthey this seems fine to me.  The set approach was based on the assumption that we would be passing simple (hashable) key/value pairs.  I will note that i've run into some edge-cases where ansible does some weird mangling from YAML to Python dicts so your mileage may vary when trying to put/post complex values